decisions/0003-files-in-JSON-output.md

```diff
@@ -0,0 +1,112 @@
+# 3. Files in JSON Output
+
+Date: 2019-08-27
+
+## Status
+
+🤔 Proposed
+
+## Context
+
+When users submit forms created by Form Builder the data from the user needs to
+be handed over to other systems for processing. This is fulfilled by the JSON
+output generated. One category of data which can be present is that of files in
+the form of documents, images and other files users may upload as part of their
+form submission.
+
+In order to hand over these files to other systems they need to be somehow
+present in the JSON.
+
+Considerations taken on board were security, ease of implementing and ease of
+integration. We need to consider that the data is sent securely and should only
+be accessible by the intended recipient. The solution should be cost effective
+and therefore cheap to implement and maintain. Integrating systems should able
+to receive files with ease therefore able to provide cost effective solutions.
+
+We have discussed the pros and cons of 4 possible options:
+
+1. Include files in JSON document as Base64 encoded strings
+
+Pros:
+
+- Super simple with a single POST of entire payload
+- We can delete all data once submission has been handed over
+
+Cons:
+
+- Potentially a large POST body
+- Files will be Base64 encoded which will increase size of payload and
+processing
+- Clients will have to decode file
+
+2. Embed signed many use AWS S3 URL in JSON
+
+Pros:
+
+- Lightweight JSON payload
+- S3 deals with files
+
+Cons:
+
+- We must retain files for client to pick up
+- Client must pick up file within 1 week (S3 constraint)
+- We can only retain files for 28 days
+- Exposed link will lead exposed data back to us
+- If client does not pick up data within 28 days the submission files will be
+lost
+
+3. Embed signed Form Builder URL.
+
+This is similar to option 2. However, instead of embedding an S3 URL we can
+embed a custom Form Builder generated URL that we have control over. This would
+requires Form Builder to have an extra application to deal with this flow.
+
+Pros:
```

Comment:
This option has all of the pros and cons of option 2 except for `Client must pick up file within 1 week (S3 constraint)` - we should list them all here too because we're comparing across all 4 options, not just between 2 and 3.

Reply:
i have duplicated the pros and cons